packages/camera/camera_android_camerax/android/src/test/java/io/flutter/plugins/camerax/ImageCaptureTest.java

```diff
@@ -0,0 +1,211 @@
+    final Long flashMode = Long.valueOf(ImageCapture.FLASH_MODE_ON);
+    final int targetResolutionWidth = 10;
+    final int targetResolutionHeight = 50;
+    final GeneratedCameraXLibrary.ResolutionInfo resolutionInfo =
+        new GeneratedCameraXLibrary.ResolutionInfo.Builder()
+            .setWidth(Long.valueOf(targetResolutionWidth))
+            .setHeight(Long.valueOf(targetResolutionHeight))
+            .build();
+
+    imageCaptureHostApiImpl.cameraXProxy = mockCameraXProxy;
+    when(mockCameraXProxy.createImageCaptureBuilder()).thenReturn(mockImageCaptureBuilder);
+    when(mockImageCaptureBuilder.build()).thenReturn(mockImageCapture);
+
+    final ArgumentCaptor<Size> sizeCaptor = ArgumentCaptor.forClass(Size.class);
+
+    imageCaptureHostApiImpl.create(imageCaptureIdentifier, flashMode, resolutionInfo);
+
+    verify(mockImageCaptureBuilder).setFlashMode(flashMode.intValue());
+    verify(mockImageCaptureBuilder).setTargetResolution(sizeCaptor.capture());
+    assertEquals(sizeCaptor.getValue().getWidth(), targetResolutionWidth);
+    assertEquals(sizeCaptor.getValue().getHeight(), targetResolutionHeight);
+    verify(mockImageCaptureBuilder).build();
+    verify(testInstanceManager).addDartCreatedInstance(mockImageCapture, imageCaptureIdentifier);
+  }
+
+  @Test
+  public void setFlashMode_setsFlashModeOfImageCaptureInstance() {
+    final ImageCaptureHostApiImpl imageCaptureHostApiImpl =
+        new ImageCaptureHostApiImpl(mockBinaryMessenger, testInstanceManager, context);
+    final Long imageCaptureIdentifier = 85L;
+    final Long flashMode = Long.valueOf(ImageCapture.FLASH_MODE_AUTO);
+
+    testInstanceManager.addDartCreatedInstance(mockImageCapture, imageCaptureIdentifier);
+
+    imageCaptureHostApiImpl.setFlashMode(imageCaptureIdentifier, flashMode);
+
+    verify(mockImageCapture).setFlashMode(flashMode.intValue());
+  }
+
+  @Test
+  public void
+      takePicture_sendsRequestToTakePictureWithExpectedConfigurationWhenTemporaryFileCanBeCreated() {
+    final ImageCaptureHostApiImpl imageCaptureHostApiImpl =
+        spy(new ImageCaptureHostApiImpl(mockBinaryMessenger, testInstanceManager, context));
+    final Long imageCaptureIdentifier = 6L;
+    final File mockOutputDir = mock(File.class);
+    final File mockFile = mock(File.class);
+    final ImageCapture.OutputFileOptions mockOutputFileOptions =
+        mock(ImageCapture.OutputFileOptions.class);
+    final ImageCapture.OnImageSavedCallback mockOnImageSavedCallback =
+        mock(ImageCapture.OnImageSavedCallback.class);
+    final GeneratedCameraXLibrary.Result<String> mockResult =
+        mock(GeneratedCameraXLibrary.Result.class);
+
+    testInstanceManager.addDartCreatedInstance(mockImageCapture, imageCaptureIdentifier);
+    when(context.getCacheDir()).thenReturn(mockOutputDir);
+    imageCaptureHostApiImpl.cameraXProxy = mockCameraXProxy;
+    mockedStaticFile
+        .when(() -> File.createTempFile("CAP", ".jpg", mockOutputDir))
```

Comment:
If you make constants they can also be referenced here. Note that without making constants I think this test would fail if you changed the name in the implementation from CAP to anything else.